packages/axe-core-api/lib/axe/core.rb

```diff
@@ -41,13 +41,18 @@ def use_run_partial
     def assert_frame_ready
       begin
         ready = Timeout.timeout(10) {
-          @page.evaluate_script <<-JS
-            document.readyState === 'complete'
-          JS
+          wait = Selenium::WebDriver::Wait.new(:timeout => 10)
```

Comment:
This now checks `readyState` repeatedly for a `'complete'` state. Previously, the state would be checked once; given an `'interactive'` state, `assert_frame_ready` would fail and never succeed, hence #353.